### PR TITLE
readme, tools: add the doc of upgrading Loader or Syncer to DM

### PR DESCRIPTION
**Pull request by CaitinChen**
Via
tidb-tools/docs: added issue template for DM #95
tidb-tools/unify DM component name #131
tidb-tools/improve dm document #132
@lilin90 @csuzhangxc @GregoryIan @morgo PTAL @_@

### CODE REVIEW

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+category: tools
+---
+
+# Upgrade Loader or Syncer to Data Migration
+
+This document introduces how to upgrade Loader or Syncer to DM (Data Migration).
+
+## Upgrade Loader to Data Migration
+
+Loader is used to import the full data that is exported from [mydumper](https://github.com/pingcap/docs/blob/master/tools/mydumper.md) to TiDB.
```
Comment (kennytm) [inline tools/upgrade-loader-or-syncer-to-dm.md]: fix link

Comment (CaitinChen) [reply tools/upgrade-loader-or-syncer-to-dm.md]: @kennytm What's wrong with this link?

Comment (kennytm) [reply tools/upgrade-loader-or-syncer-to-dm.md]: It should point to `../tools/mydumper.md` not `github.com`?

Comment (CaitinChen) [reply tools/upgrade-loader-or-syncer-to-dm.md]: @kennytm Yes, sir @_@

Comment (csuzhangxc) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
Loader is used to load the full data that is dumped from [mydumper](https://github.com/pingcap/docs/blob/master/tools/mydumper.md) to TiDB.
```

loader do *load*, and mydumper do *dump* ?

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| `pattern-schema` | `schema-pattern` |
+| `pattern-table` | `table-pattern` |
+| `do-db` |	`do-dbs` of `black-white-list`. The filtering feature of databases has been refactored. For detailed configuration, see `do-dbs` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). |  
+| `do-table` | `do-tables` of `black-white-list`. The filtering feature of tables has been refactored. For detailed configuration, see `do-tables` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). |  
+| `ignore-db` | `ignore-dbs` of `black-white-list`. The filtering feature of databases has been refactored. For detailed configuration, see `ignore-dbs` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). |
+| `ignore-table` | `ignore-tables` of `black-white-list`. The filtering feature of tables has been refactored. For detailed configuration, see `ignore-tables` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). |
+| `rm-checkpoint` | Deprecated. A configuration option with a similar feature is `remove-previous-checkpoint`. |
```
Comment (csuzhangxc) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
| `rm-checkpoint` | Deprecated. A configuration option with a similar feature is `remove-meta`. |
```

we have changed this item to `remove-meta`, but forgot to update the CN doc.

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+| `password` | No corresponding option. Fixed when DM-worker is deployed. |
+| `threads` | `threads` of `mydumper` |
+| `chunk-filesize` | `chunk-filesize` of `mydumper` |
+| `skip-tz-utc` | `skip-tz-utc` of `mydumper` |
+
+Other options are specified using `extra-args` of `mydumper`. Their usage in DM is the same as that in mydumper.
+
+## Upgrade Syncer to Data Migration
+
+Syncer is a tool used to import data incrementally. The task Syncer executes is corresponding to the synchronization task with `incremental``task-mode` in DM. The `syncer` processing unit feature of dm-worker in DM is corresponding to the Syncer feature.
```
Comment (csuzhangxc) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
Syncer is a tool used to import data incrementally. The task Syncer executes is corresponding to the synchronization task with `incremental` `task-mode` in DM. The `syncer` processing unit feature of dm-worker in DM is corresponding to the Syncer feature.
```

**Comment by csuzhangxc** [comment]
rest LGTM

**Comment by CaitinChen** [comment]
@csuzhangxc @kennytm Thanks ^_*

**Comment by csuzhangxc** [comment]
LGTM

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+| `rm-checkpoint` | Deprecated. A configuration option with a similar feature is `remove-meta`. |
+
+Taking the configuration options in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md) as examples, the corresponding relationship of task configuration options between mydumper and DM is as follows:
+
+| Configuration in mydumper | Corresponding configuration in DM |
+| ---------------------- | --------------------------------- |
+| `host` | No corresponding option. Fixed when DM-worker is deployed. |
+| `port` | No corresponding option. Fixed when DM-worker is deployed. |
+| `user` | No corresponding option. Fixed when DM-worker is deployed. |
+| `password` | No corresponding option. Fixed when DM-worker is deployed. |
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]: Is "Fixed" the meaning we want to express? Perhaps "Configured" or "Confirmed" or some other words? @csuzhangxc @CaitinChen

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+category: tools
+---
+
+# Upgrade Loader or Syncer to Data Migration
+
+This document introduces how to upgrade Loader or Syncer to DM (Data Migration).
+
+## Upgrade Loader to Data Migration
+
+Loader is used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
Loader is a tool used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
```

Comment (lilin90) [reply tools/upgrade-loader-or-syncer-to-dm.md]: Keep it consistent with the following Syncer.

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+| `ignore-db` | `ingore-dbs` of `black-white-list`. The filtering feature of databases has been refactored. For detailed configuration, see `ignore-dbs` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). | 
+| `ignore-table` | `ignore-tables` of `black-white-list`. The filtering feature of tables has been refactored. For detailed configuration, see `ignore-tables` in [Data Migration Task Configuration File](../tools/dm-task-configuration-file-intro.md). |
+| `skip-ddls` | Deprecated. Use `filters`. |
+| `skip-sqls` | Deprecated. Use `filters`. |
+| `skip-events` | Deprecated. Use `filters`. |
+| `skip-dmls` | Deprecated. Use `filters`. |
+| `route-rules` | `route-rules`|
+| `pattern-schema` | `schema-pattern` |
+| `pattern-table` | `table-pattern` |
+| `from` | `config` of `mysql-instances`. Should be consistent with the upstream MySQL information during the DM-worker deployment. |
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
| `from` | `config` of `mysql-instances`. Keep it consistent with the upstream MySQL information during the DM-worker deployment. |
```

Comment (lilin90) [reply tools/upgrade-loader-or-syncer-to-dm.md]: The word "should" is a little bit too weak. Using an imperative sentence is clearer.

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+## Upgrade Loader to Data Migration
+
+Loader is used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
+
+When the `task-mode` of the task DM executes is `full`, DM automatically uses `dumper` to export data and then uses `loader` to import the data.
+
+To upgrade Loader to DM, you only need to generate the corresponding task configuration file and use dmctl to execute the task.
+
+### Change configuration
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]: Keep it consistent with that of Syncer.

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+This document introduces how to upgrade Loader or Syncer to DM (Data Migration).
+
+## Upgrade Loader to Data Migration
+
+Loader is used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
+
+When the `task-mode` of the task DM executes is `full`, DM automatically uses `dumper` to export data and then uses `loader` to import the data.
+
+To upgrade Loader to DM, you only need to generate the corresponding task configuration file and use dmctl to execute the task.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]: @csuzhangxc Could you please provide the concrete steps (like 1, 2, 3, ...) that users can directly follow? This one sentence is ambiguous. Users might still be not clear about how to go ahead.

Later, the dmctl link will be added to this doc after that PR is merged.

**Comment by lilin90** [comment]
@csuzhangxc Please check some of my comments. Thanks!

File: tools/upgrade-loader-or-syncer-to-dm.md (hunk 4 above)
Comment (csuzhangxc) [reply tools/upgrade-loader-or-syncer-to-dm.md]: maybe "Configured" is better

File: tools/upgrade-loader-or-syncer-to-dm.md (hunk 8 above)
Comment (csuzhangxc) [reply tools/upgrade-loader-or-syncer-to-dm.md]: 1. deploy DM cluster
2. generate the corresponding task configuration file
3. use dmctl to `start-task`

**Comment by csuzhangxc** [comment]
@lilin90 checked

**Comment by lilin90** [comment]
@CaitinChen Please update this document according to @csuzhangxc's comments. By the way, some comment is ignored. Please also check and update. Thanks!

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

**Comment by CaitinChen** [comment]
@lilin90 PTAL again. Thanks.

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+Loader is a tool used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
+
+When the `task-mode` of the task DM executes is `full`, DM automatically uses `dumper` to dump data and then uses `loader` to load the data.
+
+To upgrade Loader to DM, perform the following steps:
+
+1. [Deploy the DM cluster](../tools/data-migration-deployment.md).
+2. Generate the corresponding task configuration file.
+3. Use dmctl to start the task.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
3. [Use dmctl to start the task](../tools/data-migration-manage-task.md#create-the-data-synchronization-task).
```

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+## Upgrade Loader to Data Migration
+
+Loader is a tool used to load the full data that is dumped from [mydumper](../tools/mydumper.md) to TiDB.
+
+When the `task-mode` of the task DM executes is `full`, DM automatically uses `dumper` to dump data and then uses `loader` to load the data.
+
+To upgrade Loader to DM, perform the following steps:
+
+1. [Deploy the DM cluster](../tools/data-migration-deployment.md).
+2. Generate the corresponding task configuration file.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
2. Refer to [Loader configuration change](#loader-configuration-change) and generate the corresponding task configuration file.
```

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+3. Use dmctl to start the task.
+
+### Configuration change
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
### Loader Configuration change 
```

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+Syncer is a tool used to import data incrementally. The task Syncer executes is corresponding to the synchronization task with `incremental` `task-mode` in DM. The `syncer` processing unit feature of dm-worker in DM is corresponding to the Syncer feature.
+
+To upgrade Syncer to DM, perform the following steps:
+
+1. [Deploy the DM cluster](../tools/data-migration-deployment.md).
+2. Generate the corresponding task configuration file.
+3. Use dmctl to start the task.
+
+### Configuration change
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
### Syncer Configuration change
```

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+Other options are specified using `extra-args` of `mydumper`. Their usage in DM is the same as that in mydumper.
+
+## Upgrade Syncer to Data Migration
+
+Syncer is a tool used to import data incrementally. The task Syncer executes is corresponding to the synchronization task with `incremental` `task-mode` in DM. The `syncer` processing unit feature of dm-worker in DM is corresponding to the Syncer feature.
+
+To upgrade Syncer to DM, perform the following steps:
+
+1. [Deploy the DM cluster](../tools/data-migration-deployment.md).
+2. Generate the corresponding task configuration file.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
2. Refer to [Syncer configuration change](#syncer-configuration-change) and generate the corresponding task configuration file.
```

Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
2. Refer to [Syncer configuration change](#syncer-configuration-change) and generate the corresponding task configuration file.
```

File: tools/upgrade-loader-or-syncer-to-dm.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+3. Use dmctl to start the task.
```
Comment (lilin90) [inline tools/upgrade-loader-or-syncer-to-dm.md]:
```suggestion
3. [Use dmctl to start the task](../tools/data-migration-manage-task.md#create-the-data-synchronization-task).
```

**Pull request merged**